### PR TITLE
samples: Bluetooth: Audio sample board files cleanup

### LINKED ISSUE #45543: Build samples/bluetooth/broadcast_audio_sink raises an error
**Describe the bug**
Direct build samples/bluetooth/broadcast_audio_sink for nrf52840dk_nrf52840

**To Reproduce**

cd zephyrproject/zephyr
west build -b nrf52840dk_nrf52840 -p auto -d 52840 .\samples\bluetooth\broadcast_audio_sink\

**Expected behavior**
Is this a bug or does this demo not support nrf52840dk_nrf52840?

**Logs and console output**
warning: default value 1 on BT_CTLR_ADV_SYNC_SET (defined at subsys/bluetooth\controller\Kconfig.ll_sw_split:182) clamped to 0 due to being outside the active range ([1, 0])

error: Aborting due to Kconfig warnings

**Environment (please complete the following information):**
 - OS: Windows10
 - Toolchain: arm_none_eabi/9_2020_q2_update
 - Commit: 3.0.99

### PR DESCRIPTION
**Pull request by Thalley**
Clean up some of the board specific configuration files.
Removed all CONFIG_BT_CTLR* configs from the nRF5340
appcore conf files as that does not compile the controller.

Ensured that the ISO buffer in the controller for
nRF52840 can contain SDUs from all BAP LC3 minimum
requirements, as the controller does not yet
support segmentation of SDUs.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

fixes https://github.com/zephyrproject-rtos/zephyr/issues/45543

### CODE REVIEW

**Review by jhedberg — DISMISSED** [review]

File: samples/bluetooth/unicast_audio_client/boards/nrf52840dk_nrf52840.conf
Comment (Casper-Bonde-Bose) [inline samples/bluetooth/unicast_audio_client/boards/nrf52840dk_nrf52840.conf:11]: I did not test LC3 can run on the nrf52 device with regards to CPU resources and RAM.

**Review by Casper-Bonde-Bose — DISMISSED** [review]
LGTM if someone else verified LC3 can run on the nRF5280

File: samples/bluetooth/unicast_audio_client/boards/nrf52840dk_nrf52840.conf
Comment (Thalley) [reply samples/bluetooth/unicast_audio_client/boards/nrf52840dk_nrf52840.conf:11]: Hmm, good point. Didn't actually test that, but can give it a try before merging this :)

Comment (Thalley) [reply samples/bluetooth/unicast_audio_client/boards/nrf52840dk_nrf52840.conf]: Hmm, actually I do not have a way of testing LC3 on the nRF52840 at this point, so will remove the LC3 from the board files.

File: samples/bluetooth/broadcast_audio_sink/boards/nrf52840dk_nrf52840.conf
Comment (cvinayak) [inline samples/bluetooth/broadcast_audio_sink/boards/nrf52840dk_nrf52840.conf:7]:
```suggestion
CONFIG_BT_CTLR_SYNC_ISO_PDU_LEN_MAX=155
```

These are only needed for synchronized receiver, not required for broadcaster.

File: samples/bluetooth/unicast_audio_server/boards/nrf52840dk_nrf52840.conf
Comment (cvinayak) [inline samples/bluetooth/unicast_audio_server/boards/nrf52840dk_nrf52840.conf:5]:
```suggestion
```

File: samples/bluetooth/unicast_audio_client/boards/nrf52840dk_nrf52840.conf
Comment (cvinayak) [inline samples/bluetooth/unicast_audio_client/boards/nrf52840dk_nrf52840.conf:5]:
```suggestion
```

**Review by cvinayak — CHANGES_REQUESTED** [review]

**Review by asbjornsabo — DISMISSED** [review]
LGTM

**Review by cvinayak — DISMISSED** [review]

**Comment by Thalley** [comment]
Added commit to increase scan and adv data sizes.

**Comment by Thalley** [comment]
Added commit to increase scan and adv data sizes.

File: samples/bluetooth/broadcast_audio_sink/boards/nrf52840dk_nrf52840.conf
Comment (cvinayak) [inline samples/bluetooth/broadcast_audio_sink/boards/nrf52840dk_nrf52840.conf:8]: Should it be that 252 bytes be supported?

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: samples/bluetooth/broadcast_audio_sink/boards/nrf52840dk_nrf52840.conf
Comment (Thalley) [reply samples/bluetooth/broadcast_audio_sink/boards/nrf52840dk_nrf52840.conf:8]: Chose 251 specifically as I set `CONFIG_BT_CTLR_ADV_DATA_LEN_MAX=251` because that is largest extended advertising value for a single HCI command (and surely is enough for this sample, and I think it makes more sense if the two values in the pair of samples are equal. 

I did consider 252 to support the maximum single periodic advertising report, but felt that the 251 would be better to cover both extended and periodic advertising

**Review by cvinayak — DISMISSED** [review]

**Comment by Thalley** [comment]
Rebased to solve merge conflicts

**Review by cvinayak — DISMISSED** [review]

File: samples/bluetooth/broadcast_audio_source/boards/nrf52840dk_nrf52840.conf
Comment (cvinayak) [inline samples/bluetooth/broadcast_audio_source/boards/nrf52840dk_nrf52840.conf:5]:
```suggestion
# Zephyr Controller tested maximum advertising data that can be set in a single HCI command
CONFIG_BT_CTLR_ADV_DATA_LEN_MAX=191
```

Comment (Thalley) [reply samples/bluetooth/broadcast_audio_source/boards/nrf52840dk_nrf52840.conf:5]: Hmm, so we don't always support The maximum length of a single HCI command? 

In the BSIM tests we actually set `CONFIG_BT_CTLR_ADV_DATA_LEN_MAX=255` I see, so isn't it tested at 251 already?

Comment (cvinayak) [reply samples/bluetooth/broadcast_audio_source/boards/nrf52840dk_nrf52840.conf:5]: If a periodic advertising is associated and/or a channel classification is requested the common extended header format size increases, hence controller is tested only to 191 bytes as supported AD data length max.

The `bsim_test_audio` test should reduce `CONFIG_BT_CTLR_ADV_DATA_LEN_MAX` to 191 bytes. Hope 191 is sufficient for audio usecases (otherwise controller has to support AD data fragmentation into multiple on-air PDU).

**Review by cvinayak — APPROVED** [review]

**Review by Casper-Bonde-Bose — APPROVED** [review]

**Comment by Thalley** [comment]
@carlescufi Can you please merge this? Will close 2 opened issues.

**Pull request merged**
